src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs

```diff
@@ -580,6 +575,166 @@ void Method1({symbolType} x , {symbolType} y)
             }.RunAsync();
         }
 
+        [Fact]
+        public async Task CompareSymbolFromInstanceEqualsWithConditionalAccess_VisualBasic()
+        {
+            var source = @"
+Imports Microsoft.CodeAnalysis
+Class TestClass
+    Sub Method1(x As ISymbol, y As ISymbol)
+        If x?[|.Equals(y)|] Then Exit Sub
+    End Sub
+End Class
+";
+
+            var fixedSource = @"
+Imports Microsoft.CodeAnalysis
+Class TestClass
+    Sub Method1(x As ISymbol, y As ISymbol)
+        If SymbolEqualityComparer.Default.Equals(x, y) Then Exit Sub
+    End Sub
+End Class
+";
+
+            await new VerifyVB.Test
+            {
+                TestState = { Sources = { source, SymbolEqualityComparerStubVisualBasic } },
+                FixedState = { Sources = { fixedSource, SymbolEqualityComparerStubVisualBasic } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task CompareSymbolFromInstanceEqualsWithNullConditionalAccess_CSharp()
+        {
+            var source = @"
+using Microsoft.CodeAnalysis;
+class TestClass
+{
+    void Method1(ISymbol x, ISymbol y)
+    {
+        if (x?[|.Equals(y)|] == true) return;
+    }
+}
+";
+
+            var fixedSource = @"
+using Microsoft.CodeAnalysis;
+class TestClass
+{
+    void Method1(ISymbol x, ISymbol y)
+    {
+        if (SymbolEqualityComparer.Default.Equals(x, y) == true) return;
+    }
+}
+";
+
+            await new VerifyCS.Test
+            {
+                TestState = { Sources = { source, SymbolEqualityComparerStubCSharp } },
+                FixedState = { Sources = { fixedSource, SymbolEqualityComparerStubCSharp } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task CompareSymbolFromInstanceEqualsWithChainConditionalAccess_VisualBasic()
+        {
+            var source = @"
+Imports Microsoft.CodeAnalysis
+
+Class A
+    Public b As B
+End Class
+
+Class B
+    Public s As ISymbol
+End Class
+
+Class TestClass
+    Sub Method1(a As A, s As ISymbol)
+        If a?.b?.s?[|.Equals(s)|] Then Exit Sub
+    End Sub
+End Class
+";
+
+            var fixedSource = @"
+Imports Microsoft.CodeAnalysis
+
+Class A
+    Public b As B
+End Class
+
+Class B
+    Public s As ISymbol
+End Class
+
+Class TestClass
+    Sub Method1(a As A, s As ISymbol)
+        If SymbolEqualityComparer.Default.Equals(a?.b?.s, s) Then Exit Sub
+    End Sub
+End Class
+";
+
+            await new VerifyVB.Test
+            {
+                TestState = { Sources = { source, SymbolEqualityComparerStubVisualBasic } },
+                FixedState = { Sources = { fixedSource, SymbolEqualityComparerStubVisualBasic } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task CompareSymbolFromInstanceEqualsWithChainNullConditionalAccess_CSharp()
```

Comment:
This is not currently passing. I'll give it a look when I get sometime (the new college semester just started, so might not be very soon). Feel free if anyone would like to give it a look before I get to it.